zdevito/terra
# Support for PUC Lua

This branch adds support for compiling Terra against PUC Lua (i.e. the original Lua implementation). Support for LuaJIT is preserved and existing users should be unaffected. Currently Lua version 5.1 is required.

To test, build as follows:

```
TERRA_USE_PUC_LUA=1 LUA_VERSION=5.1.5 make
```

The current support is sufficient to define Terra functions, and to call `terralib.saveobj`. Directly calling Terra tasks from within the same process is not supported. The limiting factor here is that the `ffi` library for Lua is not portable, so this support needs to be reimplemented in a portable way. In theory, this should be doable by JIT'ing the necessary glue code from Terra, but this is not yet implemented.

This branch is most beneficial on architectures like PPC64le, where LuaJIT is currently unsupported.
